### PR TITLE
[bench] Also upload the raw timing files, etc

### PR DESCRIPTION
**Pull request by JasonGross**
**Kind:** infrastructure

This allows us to access the raw timing logs, etc.  It might be too much data, so maybe we only want, say, `**/*.timing`, `**/*.log`, and `**/*.csv` (the last two for the perf repos)?

This will, e.g., give @ppedrot the ability to get better per-file timing info from `coq-performance-tests` and `coq-engine-bench` and `coq-rewriter-perf-SuperFast`.

### CODE REVIEW

**Comment by ppedrot** [comment]
What's the expected size of this? It does contain a lot of binary data, right? I think it'd be better to only keep a small subset of this.

**Comment by JasonGross** [comment]
@ppedrot Yeah, expected size is probably hundreds if not thousands of MB.  I've pushed a commit to make it much lighter.

File: dev/bench/gitlab.sh
Comment (SkySkimmer) [inline dev/bench/gitlab.sh:473]: If you want to list files you need -a

Comment (JasonGross) [reply dev/bench/gitlab.sh]: Ah, thanks, fixed

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
I am not really convinced by the amount of stuff we now store in the artifacts. The listing file is huge and not very informative, and the logs directory contains mostly redundant data that is irrelevant for bench analysis...

**Comment by JasonGross** [comment]
If you prefer, I can drop the listing file.  I thought it might be useful so that if we end up wanting more artifacts in the future, it wouldn't be hard to know what's available and where.  Shall I drop it?

**Comment by ppedrot** [comment]
I don't really have a strong opinion, but I do find that the bench artifacts shouldn't weight more than a handful of megabytes. Until recently it was about 23Mb once zipped, which seems already quite a lot. I'd like to see how it fares with this PR, but for that we need to get a full run.

**Comment by ppedrot** [comment]
@JasonGross if you rebase this I'll merge right after. We'll assess the effects on the fly.

**Comment by JasonGross** [comment]
@ppedrot Sorry for the delay, now rebased, and thanks!

**Pull request merged**
